src/gateway.c

```diff
@@ -1286,6 +1286,55 @@ static int handle_weight(struct handle *req)
 	return 0;
 }
 
+static void leaderGenexecCb(struct exec *exec, int status)
+{
+	struct gateway *g = exec->data;
+	struct handle *req = g->req;
+	sqlite3_stmt *stmt = g->stmt;
+
+	g->req = NULL;
+	g->stmt = NULL;
+
+	if (status != SQLITE_DONE && status != SQLITE_ROW) {
+		assert(g->leader != NULL);
+		failure(req, status, error_message(g->leader->conn, status));
+		sqlite3_reset(stmt);
+		return;
+	}
+	query_batch(stmt, req, status);
```

Comment:
I think it's confusing that we pass a `status` to a function parameter with the name `initial`. `initial` sounds like a bool.